notebooks/py/MERFISH.py

```diff
@@ -0,0 +1,252 @@
+# EPY: END markdown
+
+# EPY: START code
+# Display all the data in an interactive pop-up window. Uncomment to have this version work.
+
+# %gui qt5
+# display(imgs)
+
+# Display a single plane of data
+single_plane = imgs.sel({Axes.ROUND: 0, Axes.CH: 0, Axes.ZPLANE: 0})
+single_plane = single_plane.xarray.squeeze()
+plt.figure(figsize=(7,7))
+plt.imshow(single_plane, cmap='gray')
+plt.title('Round: 0, Chanel:0')
+plt.axis('off');
+# EPY: END code
+
+# EPY: START markdown
+### Filter and scale raw data before decoding into spatially resolved gene expression
+# EPY: END markdown
+
+# EPY: START markdown
+#A a high pass filter is used to remove background signal, which is typically of a low frequency. This serves to remove autoflourescence, thus enhancing the ability to detect the RNA molecules.
+# EPY: END markdown
+
+# EPY: START code
+from starfish.image import Filter
+ghp = Filter.GaussianHighPass(sigma=3)
+high_passed = ghp.run(imgs, verbose=True, in_place=False)
+# EPY: END code
+
+# EPY: START markdown
+#The below algorithm deconvolves out the point spread function (PSF) introduced by the microcope. The goal of deconvolution is to enable the resolution of more spots, especially in high transcript density regions of the data. For this assay, the PSF is well approximated by a 2D isotropic gaussian with standard deviation (sigma) of 2. This The number of iterations (here 15) is an important parameter that needs careful optimization.
+# EPY: END markdown
+
+# EPY: START code
+from starfish.types import Clip
+dpsf = Filter.DeconvolvePSF(num_iter=15, sigma=2, clip_method=Clip.SCALE_BY_CHUNK)
+deconvolved = dpsf.run(high_passed, verbose=True, in_place=False)
+# EPY: END code
+
+# EPY: START markdown
+#The data for this assay are already registered across imaging rounds. Despite this, individual RNA molecules may still not be perfectly aligned across imaging rounds. This is crucial in order to read out a measure of the itended barcode (across imaging rounds) in order to map it to the codebook. To solve for potential mis-alignment, the images can be blurred with a 1-pixel Gaussian kernel. The risk here is that this will obfuscate signals from nearby molecules, thus potentially working against the deconvolution step from the previous cell!
+# EPY: END markdown
+
+# EPY: START code
+glp = Filter.GaussianLowPass(sigma=1)
+low_passed = glp.run(deconvolved, in_place=False, verbose=True)
+# EPY: END code
+
+# EPY: START markdown
+#Image intensities vary across color channels and imaging rounds. We use the author's computed scale factors to appropriately scale the data to correct for this variation. Right now we have to extract this information from the metadata and apply this transformation manually.
+# EPY: END markdown
+
+# EPY: START code
+if use_test_data:
```

Comment:
We should fix the locations of these data so they are consistent and replace the code with the ElementWiseMultiply pipeline component. This can happen in a future PR (see #1182)

Reply:
👍